service/src/service.rs

```diff
@@ -163,6 +178,32 @@ pub async fn run() -> anyhow::Result<()> {
         graph_node_query_base_url: &config.graph_node.query_url,
     });
 
+    let agreement_store: Arc<dyn AgreementStore> = Arc::new(InMemoryAgreementStore {
+        data: RwLock::new(HashMap::new()),
+    });
```

Comment:
If `InMemoryAgreementStore` implemented `Default::default()`, it would simplify the instantiation:

```suggestion
    let agreement_store: Arc<dyn AgreementStore> = Arc::new(InMemoryAgreementStore::default());
```

Reply:
it does, I forgot to update it